### ISSUE TITLE
no-wait-for-side-effects: not reporting VariableDeclaration nodes

### ISSUE THREAD

**Issue by Belco90** [testing-library/eslint-plugin-testing-library#368]
> Also, I've noticed this rule is NOT reporting this case. Is that intentional? 🤔 
> 
> 
> 
> ```
> 
>     await waitFor(() => {
> 
>       userEvent.click(button) // reported
> 
>       const a = userEvent.click(button) // NOT reported
> 
>     })
> 
> ```

Good catch. It's not intentional, but I detected it when implementing `no-unnecessary-act`, so it will be addressed there. I'll make sure to include this case in the tests.

_Originally posted by @Belco90 in https://github.com/testing-library/eslint-plugin-testing-library/issues/363#issuecomment-830643800_

This will be finally addressed separately, after #259 gets released.

**Comment by Belco90** [comment]
@zaicevas pinging you here!

**Comment by sjarva** [comment]
I'm asking this again @Belco90: I see you have assigned this to yourself, but since you haven't made a PR yet, would you mind if I submit one? 🤗

**Comment by Belco90** [comment]
@sjarva By all means! I didn't realize it was assigned to me 😓

**Comment by sjarva** [comment]
Thanks! Could you go through all issues that you have self-assigned and unassign yourself if aren't going to make a PR in the near future. (I found at least a couple of bugs, but not sure about feature issues.)

**Comment by Belco90** [comment]
Good idea! I just reviewed the issues assigned to myself and unassigned them.
